src/types/modules/module.cpp

```diff
@@ -78,22 +78,32 @@ bool Module::isInitialized() const
 	return _structure != nullptr;
 }
 
+void Module::reset(const std::string& name)
+{
+	auto itr = knownModules.find(name);
+	if (itr != knownModules.end())
+		itr->second->_structure = nullptr;
+}
+
 /**
  * Loads the module based on its name from the table of known modules.
  *
  * @param name Name of the module to load
+ * @param avastSpecific Determines which symbols to import
  *
  * @return Module if found, @c nullptr otherwise.
  */
-std::shared_ptr<Module> Module::load(const std::string& name)
+std::shared_ptr<Module> Module::load(const std::string& name, bool avastSpecific)
 {
+	if (!avastSpecific && (name == "androguard" || name == "phish")) // the androguard and phish modules are completely avast-specific
```

Comment:
I would like this to be an attribute of the module itself, like _which features you require in order to load_? This feels like a hack because information about a module is taken out from it.

Reply:
Understood, instead of specifying here names of unsupported modules we get this information out of the module itself. Implemented [here](https://github.com/avast/yaramod/pull/51/commits/591e8499aeced53445819fa2b3602219cb4645a5).